lib/authorization.rb

```diff
@@ -100,44 +100,17 @@ def hyper_tag(project)
     def associate_with_project(project)
```

Comment:
> Authorization::HyperTagMethods is staying, because that is how
objects are associated to projects, unrelated to
authorization/access control.

In version 1, we designed the project-resource association coupled with authorization. If version 2 doesn't accommodate this, I suggest we completely remove the `AccessTag`. It only seems to complicate matters. The `PostgresResource` has a `project_id` column, which is unique compared to other resources. @byucesoy wanted to experiment with having a `project_id` column instead of establishing an association through authorization. @fdr was willing to associating a single resource with multiple projects feature such as migration resources between projects. However, it appears we don't currently have a plan for a feature that leverages a single resource across multiple projects. Therefore, in my opinion, we can simplify the things by adding a `project_id` column to all resources and completely eliminate the `AccessTag`. This approach aligns with what @byucesoy implemented for `PostgresResources`, if @fdr has no objections.

Reply:
I strongly feel that most things that most resources that currently hypertag projects should use a `project_id` column instead. Having anything other than 1 project associated for these resources seems wrong.  Having zero projects seems like a leak, and having multiple projects doesn't make sense to me. You can migrate/transfer a resource between projects by changing the `project_id` value.

The one except is users <-> projects, which is a many-to-many relationship.  For that reason, you would still need AccessTag, though maybe it could be renamed and use actual foreign keys.

This PR doesn't include such changes, because they are not technically related to authorization.

Reply:
I'm supportive of having `project_id` column in the resources instead of having association. I think at some point, we were considering adding one resource to multiple projects to enable complex management options, but we didn't pursue that idea. Currently, we already have code in various places doing `projects.first`, which assumes there is only one project.

Reply:
OK.  I can work on a PR to do that after this PR is merged, unless someone else would prefer to work on it.